### ISSUE TITLE
Subqueries using let not working as expected OrientDB 3.x vs 2.2.30

### ISSUE THREAD

**Issue by misu200** [orientechnologies/orientdb#8511]
### OS: Windows 10 

I have some more complex working  queries using LET in Orient 2.3.30 that no more work in OrientDB 3.x
but I ve simplified the problem to the following test:

```
create class A extends V;
create class B extends V;
create class X extends E;								  
let $a = create vertex A;
let $b = create vertex B;
create edge X from $a to $b;
								  
select $t from A let $t=(select * from (select expand($parent.$current.out()))); 
```

I always have an empty value for $t in 3.x and the expected $t value  (e.g.  something like "$t": ["#34:0"] ) in 2.2.30 )

I think its connected to issue #8383 also

**Comment by luigidellaquila** [comment]
Hi @misu200 

I didn't try it, but I'd say in this case you expect the inner select to get records from two levels above, so it should be `$parent.$parent.$current`

Thanks

Luigi

**Comment by misu200** [comment]
Hi @luigidellaquila 

I ve changed the test to use $parent.$parent.$current instead of $parent.$current

OrientDB 3.x returns empty value for $t .... OrientDB 2.2.30 returns correct value.

Thanks,

**Comment by luigidellaquila** [comment]
Hi @misu200 

I'm pushing a fix for this problem to both develop and 3.0.x

Thanks

Luigi
